simple_graph.py

```diff
@@ -16,17 +16,25 @@ def __init__(self):
 
     def add_node(self, value):
         """Add node to Graph."""
-        self.container.setdefault(value, [])
+        if value in self.container:
+            pass
+        else:
+            self.container[value] = {}
 
     def has_node(self, value):
         """Check if node is in the graph."""
         return value in self.container
 
-    def add_edge(self, pointer_node, destination_node):
+    def add_edge(self, pointer_node, destination_node, weight=0):
         """Add edge between two nodes. If not in graph add nodes and edge."""
+        if pointer_node not in self.container:
+            self.add_node(pointer_node)
         if destination_node not in self.container:
             self.add_node(destination_node)
-        self.container.setdefault(pointer_node, []).append(destination_node)
+        if destination_node in self.container[pointer_node]:
+            pass
+        else:
```

Comment:
This full conditional is unnecessary. You could just test `if destination_node not in self.container[pointer_node]`